glusterd2/commands/volumes/volume-edit-metadata.go

```diff
@@ -0,0 +1,116 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerVolEditMetadataStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"vol-edit-metadata", editVolMetadata},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func volumeEditMetadataHandler(w http.ResponseWriter, r *http.Request) {
+
+	p := mux.Vars(r)
+	volname := p["volname"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req api.VolEditMetadataReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, err)
+		return
+	}
+
```

Comment:
Add one more validation to not allow Metadata set if key name starts with `_`. Keys starting with `_` can be reserved for internal use. Refer: https://github.com/gluster/glusterd2/pull/603/files#diff-b3cf3c6dd98c4e29314e6f748f518b21R29